#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
  contactProperties [
    ContactProperties {
      material2 "slippy"
      coulombFriction [
        0
      ]
    }
    ContactProperties {
      material2 "block"
      coulombFriction [
        0.01
      ]
    }
  ]
}
Viewpoint {
  orientation -0.15012319498306267 -0.9640500461297187 -0.21924993702476786 2.2415938802060635
  position -2.691449377721036 1.7009598137880793 -2.048984116164522
}
Background {
  skyColor [
    0.15 0.45 1
  ]
}
Floor {
  size 2.4 2.4
  appearance Asphalt {
  }
}
DEF Floor_colour Solid {
  translation 0 0.0001 0.4
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(6)"
}
DEF Floor_colour Solid {
  translation 0 0.0001 -0.4
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
  name "solid(5)"
}
DEF WALL_1 Solid {
  translation 1.2 0.05 0
  children [
    DEF WALL_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0 0.333333 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.02 0.1 2.38
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -1.2 0.05 0
  children [
    USE WALL_SHAPE
  ]
  name "solid(1)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.05 -1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(2)"
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.05 1.2
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  name "solid(3)"
  boundingObject USE WALL_SHAPE
}
TexturedBackgroundLight {
  texture "empty_office"
}
target {
  translation -0.64 0.03 0
}
target {
  translation -0.64 0.03 0.14
  name "target(4)"
}
target {
  translation 0.06 0.03 0.47
  name "target(1)"
}
target {
  translation 0.7 0.03 0
  name "target(2)"
}
target {
  translation -1 0.03 -1.01
  name "target(3)"
}
Robot {
  translation 0 0.05 0
  rotation 0 -1 0 2.78858
  children [
    Receiver {
      type "infra-red"
      channel 2
    }
    Emitter {
      type "infra-red"
      channel 1
    }
    Compass {
      yAxis FALSE
    }
    GPS {
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 1.0941561726170832e-15 -3.949756587250008e-16 1 1.570799999999999
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 4.514007528285723e-16 7.605121379177033e-17 1 1.5707999999999998
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 3.532701543875784e-16 -4.73480137477796e-16 1 1.5707999999999995
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation -5.7161073591879e-16 -3.164711799722057e-16 1 1.570799999999999
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_FRONT DistanceSensor {
      translation 0 0 0.1
      rotation 0 1 0 -1.5708053071795867
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_front"
      lookupTable [
        0 1000 0
        1.5 0 0
      ]
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "four_wheels_detect_test"
}
Robot {
  translation 0 0.84 0
  children [
    Receiver {
      type "infra-red"
      channel 1
    }
    Emitter {
      type "infra-red"
    }
  ]
  name "Creepy Observer"
  controller "CreepyObserver"
}
